### PR TITLE
[REF] project: remove iframe used in project sharing feature

### PR DESCRIPTION
**Pull request by xavierbol**
## [REM] project: remove outdated portal routes

This commit removes the routes deprecated since version 15.

## [IMP] project: remove iframe in project sharing feature

Before this commit, the project sharing feature was displayed in iframe
and so 2 bundles have to be loaded to display the feature. The first one
for the portal view (to have the main portal header) and the second one
to import the backend assets. The first one is not really needed since
it just allows to the current user to access to dropdown menu when he
clicks on his profile.

This commit removes the iframe and now only load the backend bundle
needed to load project sharing feature. This commit also reviews the
templates and review the routes to be able to refresh and keep the page
where the user was before refreshing the view.

## [IMP] project: add go back button to go back in my projects portal

Before this commit, it was not possible for the user to go back to
`/my/projects` route without altering the url.

This commit adds a button to be able to go back to `/my/projects` route.

## [IMP] project,*: move additional context for project sharing in user context

Before this commit, since the changes to be able to remove the iframe
inside project sharing feature and be able to redirect the portal user
in the right route when he refreshes the page, it was no longer possible
to pass the additional context to the first action loaded by project
sharing as it was one before. However, the `user_context` inside the
session_info is propagated to the actions, that key could contain the
additional context used to continue to know which feature is enabled
inside the project shared.

This commit moves the content of `action_context` in session_info of
project sharing feature inside `user_context` contains in session_info
to continue to propagate that additional context as before.

task-4267190

### CODE REVIEW

**Review by pkyriakou — COMMENTED** [review]
Thank you @xavierbol I didn't really have anything to comment on :shrug: LGTM :partying_face: :muscle:

File: addons/web/static/src/core/browser/router.js [hunk 1]
```diff
@@ -115,6 +115,10 @@ function pathFromActionState(state) {
     return path.join("/");
 }
 
+export function getStartUrl() {
```
Comment (xavierbol) [inline addons/web/static/src/core/browser/router.js:118]: Coucou @aab-odoo 

Is it okay for you to add that hook? 🙂 

Thanks in advance! 🙂

**Review by xavierbol — COMMENTED** [review]
robodoo rebase-ff r+

**Comment by xavierbol** [comment]
robodoo r+

(I just rebase the branch)

**Pull request closed without merging**